mettle/src/mettle.c

```diff
@@ -197,6 +199,38 @@ struct mettle *mettle(void)
 	return NULL;
 }
 
+const char * mettle_get_machine_id(void)
+{
+  struct utsname utsbuf;
+  struct dirent *data;
+  static char machine_id[MAX_PATH] = "";
+
+  if (uname(&utsbuf) == 0) {
+    strncat(machine_id, utsbuf.nodename, sizeof(machine_id) - strlen(utsbuf.nodename) - 1);
+  }
+
+  DIR *ctx = opendir("/dev/disk/by-uuid");
+
+  if (ctx) {
+    while ((data = readdir(ctx)) != NULL) {
+      if (!strcmp(data->d_name, ".") || !strcmp(data->d_name, "..")) {
```

Comment:
Disk files don't start with `.` anyway. So perhaps this line can be:
``` c
if (data->d_name[0] != '.') {
    // stuff
}
```
Then we don't even need a negative case.